aws/resource_aws_default_security_group.go

```diff
@@ -1,40 +1,188 @@
 package aws
 
 import (
+	"bytes"
 	"fmt"
 	"log"
+	"sort"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/hashcode"
 	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/ec2/finder"
 )
 
+const DefaultSecurityGroupName = "default"
+
 func resourceAwsDefaultSecurityGroup() *schema.Resource {
-	// reuse aws_security_group_rule schema, and methods for READ, UPDATE
-	dsg := resourceAwsSecurityGroup()
-	dsg.Create = resourceAwsDefaultSecurityGroupCreate
-	dsg.Delete = resourceAwsDefaultSecurityGroupDelete
-
-	// description is a computed value for Default Security Groups and cannot be changed
-	dsg.Schema["description"] = &schema.Schema{
-		Type:     schema.TypeString,
-		Computed: true,
-	}
-
-	// name is a computed value for Default Security Groups and cannot be changed
-	delete(dsg.Schema, "name_prefix")
-	dsg.Schema["name"] = &schema.Schema{
-		Type:     schema.TypeString,
-		Computed: true,
-	}
-
-	// We want explicit management of Rules here, so we do not allow them to be
-	// computed. Instead, an empty config will enforce just that; removal of the
-	// rules
-	dsg.Schema["ingress"].Computed = false
-	dsg.Schema["egress"].Computed = false
-	return dsg
+	return &schema.Resource{
+		Create: resourceAwsDefaultSecurityGroupCreate,
+		Read:   resourceAwsDefaultSecurityGroupRead,
+		Update: resourceAwsDefaultSecurityGroupUpdate,
+		Delete: resourceAwsDefaultSecurityGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		SchemaVersion: 1,
```

Comment:
We need to include the matching `MigrateState:  resourceAwsSecurityGroupMigrateState,` here as well, or it can generate errors for existing states that upgrade provider versions.